templates/agent/scripts/couchbase.py.j2

```diff
@@ -0,0 +1,158 @@
+import json
+import urlparse
+import requests
+import sys
+
+
+# Usage examples:
+# Cluster:
+
+# python couchbase.py cluster maxBucketCount int
+# python couchbase.py cluster name string
+# python couchbase.py cluster buckets.uri string
+# python couchbase.py cluster storageTotals.hdd.free int
+# python couchbase.py cluster nodes string_list
+
+# Bucket
+
+# python couchbase.py bucket names string_list
+# python couchbase.py bucket <name>  stats.op.samples.ops avg
+
+# Node
+# python couchbase.py node <hostname> interestingStats.couch_views_data_size int
+
+
+couchbase_api_endpoint = "{{ zabbix_couchbase.api_endpoint }}"
+cluster_url = urlparse.urljoin(couchbase_api_endpoint, "/pools/default/")
+buckets_url = urlparse.urljoin(couchbase_api_endpoint, "/pools/default/buckets/")
+
+
+def get_from_url(url, params={}):
+    return json.loads(requests.get(url, params=params).content)
+
+def handle_data_by_type(value, data_type, macro=None):
+    if data_type == "int":
+        return int(value)
+
+    if data_type == "string":
+        return str(value)
+
+    if data_type == "float":
+        return float(value)
+
+    # if data_type is avg, then value should be a list,
+    # and the each element in this list is numeric.
+    # then take average of values in this list.
+    if data_type == "avg":
+        assert type(value) == list
+        return sum(value) / float(len(value))
+
+    # value should be a list of string
+    if data_type == "string_list":
+        assert type(value) == list
+        return json.dumps({
+            "data": [{"%s%s%s" % ("{", macro, "}") : str(item)} for item in value]
+        })
+
+def get_data_from_keys(data, keys_string):
+    """
+    Get wanted data through layers.
+    """
+    keys = keys_string.split(".")
+    for key in keys:
+        data = data[key]
+    return data
+
+
+def handle_cluster(args):
+    # couchbase['cluster', key, type]
+    data = get_from_url(cluster_url)
+    data_type = args[1]
+    keys_string = args[0]
+
+    result = get_data_from_keys(data, keys_string)
+
+    if keys_string == "nodes":
+        result = [item['hostname'].split(":")[0] for item in result]
+        print handle_data_by_type(result, data_type, "#NODEHOST")
```

Comment:
I would really move the `discovery` items to a dedicated function - just because that's what people would expect. Not hidden in the middle of a function.